.github/workflows/upgrade-testing.yml

```diff
@@ -0,0 +1,167 @@
+name: Upgrade Tests
+
+on:
+  push:
+    branches:
+      - trunk
+    # Always test the workflow after it's updated.
+    paths:
+      - '.github/workflows/upgrade-testing.yml'
+      - '.github/workflows/upgrade-testing-run.yml'
+  pull_request:
+    # Always test the workflow when changes are suggested.
+    paths:
+      - '.github/workflows/upgrade-testing.yml'
+      - '.github/workflows/upgrade-testing-run.yml'
+  workflow_dispatch:
+    inputs:
+      new-version:
+        description: 'The version to test updating to.'
+        type: string
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name for pull requests
+  # or the commit hash for any other events.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
+jobs:
+  #
+  # Spawns upgrade testing for WordPress 6.x versions with MySQL.
+  #
+  upgrade-tests-wp-6x-mysql:
```

Comment:
I chose to make each job DB type specific because if you include both in 1 strategy matrix:
- it will definitely exceed the maximum of 256 jobs
- The list of excludes could end up being long.
Having it separate just makes it more clear when reading as well.